kpatch-build/create-diff-object.c

```diff
@@ -3676,6 +3681,206 @@ static void kpatch_create_callbacks_objname_rela(struct kpatch_elf *kelf, char *
 	}
 }
 
+static void kpatch_set_pfe_link(struct kpatch_elf *kelf)
+{
+	struct section* sec;
+	struct rela *rela;
+
+	if (!kelf->has_pfe)
+		return;
+
+	list_for_each_entry(sec, &kelf->sections, list) {
+		if (strcmp(sec->name, "__patchable_function_entries")) {
+			continue;
+		}
+
+		if (!sec->rela) {
+			continue;
+		}
+		list_for_each_entry(rela, &sec->rela->relas, list) {
+			rela->sym->sec->pfe = sec;
+		}
+	}
+}
```

Comment:
What's the purpose of this function?  Is it really possible to have multiple `__patchable_function_entries` sections?  If so, it would be a good idea to have a comment at the top.

Reply:
Yes, on more recent kernel/toolchain combinations you'll see object files with sections like:
```
readelf --wide --sections fs/proc/meminfo.o | grep ' __patchable'
  [10] __patchable_function_entries PROGBITS        0000000000000000 000eb0 000008 00 WAL  8   0  8
  [12] __patchable_function_entries PROGBITS        0000000000000000 000eb8 000008 00 WAL  1   0  8
  [14] __patchable_function_entries PROGBITS        0000000000000000 000ec0 000008 00 WAL  1   0  8
```
(Older combinations collated the entries into a section pfe / rela section pair.)